solidity-v1/dashboard/src/connectors/tally-injected.js

```diff
@@ -0,0 +1,76 @@
+import { AbstractConnector } from "./abstract-connector"
+import { WALLETS } from "../constants/constants"
+import BigNumber from "bignumber.js"
+import { UserRejectedConnectionRequestError } from "./utils"
+
+class TallyInjectedConnector extends AbstractConnector {
+  constructor() {
+    super(WALLETS.TALLY.name)    
+    this.provider = window.tally
+  }
+
+  _onAccountChanged = ([address]) => {
+    this.emit("accountsChanged", address)
+  }
+
+  _onDisconnect = () => {
+    this.emit("disconnect")
+  }
+
+  _onChainChanged = (chainId) => {
+    this.emit("chainChanged", chainId)
+  }
+
+  enable = async () => {
+    if (!this.provider) {
+      throw new Error("window.tally provider not found")
```

Comment:
If you agree with above then we should probably change the error text here. Maybe we could just use "provider not found"?

Also we should make sure that we set `autoRefresOnNetworkChange`  to false for MetaMask only so we would probably have to do some proper changes to that:

```
if (walletName === WALLETS.NAME.METAMASK) {
      // https://docs.metamask.io/guide/ethereum-provider.html#ethereum-autorefreshonnetworkchange
      this.provider.autoRefreshOnNetworkChange = false
}
```

Reply:
https://github.com/keep-network/keep-core/pull/2744/commits/3248bcf4a251f24f0c6e09faa4402b92f67cb471